pkg/apis/numaflow/v1alpha1/edge_types.go

```diff
@@ -39,7 +39,33 @@ type Edge struct {
 }
 
 type ForwardConditions struct {
-	KeyIn []string `json:"keyIn" protobuf:"bytes,1,rep,name=keyIn"`
+	// Tags used to specify tags for conditional forwarding
+	Tags *TagConditions `json:"tags" protobuf:"bytes,1,opt,name=tags"`
+}
+
+type LogicOperator string
+
+const (
+	LogicOperatorAnd LogicOperator = "and"
+	LogicOperatorOr  LogicOperator = "or"
+	LogicOperatorNot LogicOperator = "not"
+)
+
+type TagConditions struct {
+	// Operator specifies the type of operation that should be used for conditional forwarding
+	// value could be "and", "or", "not"
+	// +kubebuilder:validation:Enum=and;or;not
+	// +optional
+	Operator *LogicOperator `json:"operator" protobuf:"bytes,1,opt,name=operator"`
+	// Values tag values for conditional forwarding
+	Values []string `json:"values" protobuf:"bytes,2,rep,name=values"`
+}
+
+func (tc TagConditions) GetOperator() LogicOperator {
+	if tc.Operator == nil {
+		return LogicOperatorOr
+	}
+	return *tc.Operator
```

Comment:
Better to have a logic to return the valid operator, similar to line 87-91 in this file.

Reply:
we have Kubebuilder's Enum validation check, the only check we need is nil check

Reply:
Kubebuilder's enum validation only works when full CRD is installed.

Reply:
Okayy updated.